### PR TITLE
MINOR: [CI] Fix centos cmake path

### PR DESCRIPTION
**Pull request by assignUser**
cmake file naming convention changed sometime between 3.18 and 3.23

### CODE REVIEW

**Comment by assignUser** [comment]
@kszucs  quick review please?

**Comment by kszucs** [comment]
Does this fix any particular build failures?

**Comment by assignUser** [comment]
Not any merged ones but the workflow for [R nightly builds](https://github.com/assignUser/test-repo-a/runs/6457555112?check_suite_focus=true#step:6:363) I am currently working on fails due this if I use `-r https://github.com/apache/arrow -b master` or if I use my fork it fails in autobrew due to the ref not existing in this repo. So more of a future fix :)

File: ci/docker/centos-7-cpp.dockerfile [hunk 1]
```diff
@@ -30,7 +30,7 @@ RUN yum install -y \
 
 ARG cmake=3.23.1
 RUN wget -nv -O - https://github.com/Kitware/CMake/releases/download/v${cmake}/cmake-${cmake}-Linux-x86_64.tar.gz | tar -xzf - -C /opt
```
Comment (kszucs) [inline ci/docker/centos-7-cpp.dockerfile:32]: @assignUser could you perhaps update the install setup to place it under `/opt/cmake-${cmake}` instead, so the path is always correct regardless the cmake version?

Comment (assignUser) [reply ci/docker/centos-7-cpp.dockerfile]: Thanks for the suggestion, tested it successfully here: https://github.com/assignUser/test-repo-a/runs/6458346977?check_suite_focus=true#step:6:392

**Review by kszucs — APPROVED** [review]
Thanks Jacob!

**Pull request merged**